### PR TITLE
Hitbtc3 setLeverage

### PR DESCRIPTION
**Pull request by Dan-krm**
Added setLeverage to Hitbtc3:
```
hitbtc3.setLeverage (10, BTC/USDT:USDT, [object Object])
2022-03-31T21:34:29.945Z iteration 0 passed in 260 ms

{
  symbol: 'BTCUSDT_PERP',
  type: 'isolated',
  leverage: '10.00',
  created_at: '2022-03-31T21:34:30.346Z',
  updated_at: '2022-03-31T21:34:30.346Z',
  currencies: [
    {
      code: 'USDT',
      margin_balance: '7.000000000000',
      reserved_orders: '0',
      reserved_positions: '0'
    }
  ],
  positions: null
}
```

### CODE REVIEW

File: js/hitbtc3.js [hunk 1]
```diff
@@ -2083,6 +2083,41 @@ module.exports = class hitbtc3 extends Exchange {
...
+        }
+        if (params['margin_balance'] === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a margin_balance parameter that will transfer margin to the specified trading pair');
+        }
+        const market = this.market (symbol);
+        const amount = this.safeNumber (params, 'margin_balance');
+        if (market['type'] !== 'swap') {
+            throw new BadSymbol (this.id + ' setLeverage() supports swap contracts only');
+        }
+        if (symbol === 'BTC/USDT:USDT') {
```
Comment (kroitor) [inline js/hitbtc3.js:2099]: Can we avoid hardcoding these symbols? Let's add them to user-controllable options:

```JavaScript
describe() {
    'options': {
        'setLeverage': {
            'maxLeverage': {
                'BTC/USDT:USDT': 100,
                'ETH/USDT:USDT': 75,
                'ADA/USDT:USDT': 75,
                'SOL/USDT:USDT': 75,
                'XRP/USDT:USDT': 75,
            },
        },
    },
}

async setLeverage (...) {
     const options = this.safeValue (this.options, 'setLeverage', {});
     const maxLeverages = this.safeValue (options, 'maxLeverage', {});
     const maxLeverage = this.safeNumber (maxLeverages, symbol, 50);
     if ((leverage < 1) || (leverage > maxLeverage)) {
         throw new BadRequest (this.id + ' setLeverage() leverage should be between 1 and ' + maxLeverage.toString () + ' for ' + symbol);
     }
     // ...
}
```

File: js/hitbtc3.js [hunk 2]
```diff
@@ -302,6 +302,15 @@ module.exports = class hitbtc3 extends Exchange {
...
                     'derivatives': 'derivatives',
                 },
+                'setLeverage': {
+                    'maxLeverage': {
+                        'BTC/USDT:USDT': 100,
+                        'ETH/USDT:USDT': 75,
+                        'ADA/USDT:USDT': 75,
+                        'SOL/USDT:USDT': 75,
+                        'XRP/USDT:USDT': 75,
+                    },
```
Comment (samgermain) [inline js/hitbtc3.js:312]: You can get this information from `this.market (symbol)['limits']['leverage']['max']` so don't hard code it, also, I counted 23 swap markets in `fetchMarkets`, I don't know if that matters

Comment (kroitor) [reply js/hitbtc3.js:312]: @samgermain as far as I understood from the code, the intent was to override the max leverage for some of them, while all others default to 50, but that might need one more confirmation, indeed. @Dan-krm could you please rewrite once again it to use the max leverage limit from the market structure as @samgermain suggested?

Comment (kroitor) [reply js/hitbtc3.js:312]: @Dan-krm nvm, i'll add this edit.

Comment (Dan-krm) [reply js/hitbtc3.js:312]: Thanks for adding the edit, I'll keep this approach in mind for the future

**Pull request merged**